types/events.go

```diff
@@ -271,9 +273,603 @@ type EventSystemKilledAccount struct {
 	Topics []Hash
 }
 
-// EventTreasuryDeposit is emitted when some funds have been deposited
-type EventTreasuryDeposit struct {
+// EventAssetIssued is emitted when an asset is issued.
+type EventAssetIssued struct {
+	Phase   Phase
+	AssetID U32
+	Who     AccountID
```

Comment:
@mikiquantum do we need to use the `Address` type here and in all the other events that use account ids in order to suppor chains that use indices pallet?

Reply:
I believe most of the AccountIDs should be parsed correctly independently of having indices pallet or not, since it is the basic form of a public address.
We already have AccountID parsed correctly in other current events like transfer. I expect should work the same.
For events that are triggered by the Identity pallet, I am not completely certain though.

Reply:
You should be right – we only have the issue with the `Address` type. 

The big confusion comes from the fact that most types can be changed for your runtime, as long as they implement the correct interfaces, e. g. `Address` could also become an arbitrary struct.

Reply:
All good for now though